File: beginner/1117.c
#include <stdio.h>

int readScore() {
	double d;
	scanf("%lf", &d);
	if (d >= 0 && d <= 10)
		return d;
	return -1;
}

int main(void) {
	
	double a;
	double b;
	
	a = readScore();
	
	while (a == -1) {
		printf("nota invalida\n");
		a = readScore();
	}
	
	b = readScore();
	
	while (b == -1) {
		printf("nota invalida\n");
		b = readScore();
	}
	
	printf("media = %.2lf\n", (a + b) / 2.0);
}

File: beginner/1117.py
def get_nota():
    nota = float(input())
    if nota < 0 or nota > 10:
        return -1
    return nota

qtd = 0
soma = 0

while qtd < 2:
    atual = get_nota()

    while atual == -1:
        print("nota invalida")
        atual = get_nota()

    soma += atual
    qtd += 1
    
print("media = %.2f" % (soma/qtd))
